src/components/program-progress/SubscriptionExpiringWarningModal.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useContext } from 'react';
+import moment from 'moment';
+import {
+  ActionRow, Button, MailtoLink, StandardModal,
+} from '@edx/paragon';
+import { AppContext } from '@edx/frontend-platform/react';
+
+import PropTypes from 'prop-types';
+import { UserSubsidyContext } from '../enterprise-user-subsidy';
+
+export const SUBSCRIPTION_EXPIRING_MODAL_TITLE = 'Your subscription access expires before the program will finish';
+
+const SubscriptionExpirationWarningModal = ({
+  isSubscriptionExpiringWarningModalOpen,
+  onSubscriptionExpiringWarningModalClose,
+}) => {
+  const {
+    enterpriseConfig: { name, contactEmail },
+  } = useContext(AppContext);
+
+  const { subscriptionPlan } = useContext(UserSubsidyContext);
+  const { expirationDate } = subscriptionPlan;
+  if (isSubscriptionExpiringWarningModalOpen === false) { return null; }
```

Comment:
As suggested in the above component, it would make more sense to make it a part of return method